web/packages/teleport/src/components/hooks/useUrlFiltering/useUrlFiltering.ts

```diff
@@ -42,23 +42,44 @@ export function useUrlFiltering(
   initialParams: Partial<ResourceFilter>
 ): UrlFilteringState {
   const { search, pathname } = useLocation();
-  const [params, setParams] = useState<ResourceFilter>({
-    ...initialParams,
-    ...getResourceUrlQueryParams(search),
-  });
 
   function replaceHistory(path: string) {
     history.replace(path);
   }
 
   function setSort(sort: SortType) {
-    setParams({ ...params, sort });
+    replaceHistory(
+      encodeUrlQueryParams({
+        pathname,
+        searchString: params.search || params.query,
+        sort: { ...params.sort, ...sort },
+        kinds: params.kinds,
+        isAdvancedSearch: true,
```

Comment:
I don't think we should always set it to `true`. I believe it should be `!!params.query`. Same for `setParams`.